tokio/src/task/mod.rs

```diff
@@ -47,12 +52,12 @@ pub(crate) struct Task<S: 'static> {
     _p: PhantomData<S>,
 }
 
-unsafe impl<S: Send + Sync + 'static> Send for Task<S> {}
+unsafe impl<S: ScheduleSend + Send + Sync + 'static> Send for Task<S> {}
```

Comment:
I'm not sure I follow this additional requirement. Shouldn't it be fine for `Task<S>` to be `Send` even if `S` isn't `ScheduleSend`?